client/client-multi/client-c/CMakeLists.txt

```diff
@@ -16,56 +16,59 @@
 
 cmake_minimum_required(VERSION 2.8.8)
 
-project (Kaa-c)
+project(Kaa-c)
 enable_language(C)
 
+# Expose Kaa SDK directory to all modules
+set(KAA_SDK_DIR ${CMAKE_CURRENT_LIST_DIR})
+
 # Path to CMake's scripts.
-set(CMAKE_MODULE_PATH 
-        ${CMAKE_MODULE_PATH} 
+set(CMAKE_MODULE_PATH
+        ${CMAKE_MODULE_PATH}
         ${CMAKE_CURRENT_SOURCE_DIR}/Modules
-    )
+        )
 
 #
 # Sets common compilation flags.
-# 
-# NOTE: Platform specific compilation flags should be set 
+#
+# NOTE: Platform specific compilation flags should be set
 # in the corresponding platform CMake script(s).
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC -Wall")
```

Comment:
Why should it be moved? The flag still should be applied even if a compiler is clang.